src/apiCache.ts

```diff
@@ -0,0 +1,81 @@
+import { brotliCompress, brotliDecompress } from 'zlib';
+import { RedisClient } from 'redis';
+import { promisify } from 'util';
+import { Request } from 'express';
+import { toPairs } from 'lodash';
+import * as flatted from 'flatted';
+import { ERedisFlag, IApiCacheConfiguration } from './types';
+import { defaultConfiguration } from './config';
+
+/**
+ * ApiCache class.
+ */
+export class ApiCache {
+  private readonly redis: RedisClient;
+  private config: IApiCacheConfiguration;
+  public redisSetAsync: (key: string, value: string, flag?: ERedisFlag, expirationInMS?: number) => Promise<unknown>;
+  public redisGetAsync: (key: string) => Promise<string | null>;
+  private readonly compressAsync: (buffer: Buffer) => Promise<Buffer>;
+  private readonly decompressAsync: (buffer: Buffer) => Promise<Buffer>;
+
+  /**
+   * @description Constructor.
+   * @param {RedisClient} redis
+   * @param {IApiCacheConfiguration} config
+   */
+  constructor(redis: RedisClient, config?: IApiCacheConfiguration) {
+    this.redis = redis;
+    this.config = { ...defaultConfiguration, ...config };
+    this.redisSetAsync = promisify(this.redis.set).bind(this.redis);
+    this.redisGetAsync = promisify(this.redis.get).bind(this.redis);
+    this.compressAsync = promisify(brotliCompress);
+    this.decompressAsync = promisify(brotliDecompress);
+  }
+
+  /**
+   * @description Builds the redis key associated with the corresponding request.
+   * @param {Request} req Express request
+   * @private
+   */
+  private buildKey(req: Request): string {
+    const generatedQuery = toPairs(req.query)
+      .map((pair: [string, unknown]) => pair.join(''))
```

Comment:
string instead of unknown no?

Reply:
the compiler refuses the type [string, string] it expects [string,unknown]